scheduler/graphql_mid/schema.py

```diff
@@ -83,6 +87,32 @@ def change_origin(self, new_origin: SOrigin, mode: SchedulerModes) -> ChangeOrig
         sources.set_origin(new_origin)
         return ChangeOriginSuccess(from_origin=old, to_origin=str(new_origin))
 
+    @strawberry.mutation
+    def add_weather_change(self, new_weather_change: NewWeatherChangeInput) -> EventsAddedResponse:
+        try:
+            new_cc = new_weather_change.new_CC
+            new_sb = new_weather_change.new_SB
+            new_wv = new_weather_change.new_WV
+            new_iq = new_weather_change.new_IQ
+            c = Conditions(cc=CloudCover[new_cc.name] if new_cc else None,
+                           sb=SkyBackground[new_sb.name] if new_sb else None,
+                           wv=WaterVapor[new_wv.name] if new_wv else None,
+                           iq=ImageQuality[new_iq.name] if new_iq else None)
```

Comment:
I'm still concerned about these being `None` if they aren't defined because the types in `Conditions` aren't `Optional`s so we should never be setting them to `None`? We should think of an alternate way to do this.